ktlint-gradle-plugin/src/main/kotlin/io/github/usefulness/KtlintGradlePlugin.kt

```diff
@@ -34,8 +34,8 @@ public class KtlintGradlePlugin : Plugin<Project> {
         val ktlintConfiguration = createKtlintConfiguration(pluginExtension)
         val ruleSetConfiguration = createRuleSetConfiguration(ktlintConfiguration)
         val reportersConfiguration = createReportersConfiguration(ktlintConfiguration)
-        val recognisedEditorConfigs = generateSequence(project) { it.parent }
-            .map { it.layout.projectDirectory.file(".editorconfig").asFile }
+        val recognisedEditorConfigs = generateSequence(projectDir) { if (it == rootProject.projectDir) null else it.parentFile }
+            .map { it.resolve(".editorconfig") }
```

Comment:
Note that this will now traverse _directories_ and not _projects_. Previously if projects were set up with custom directories, the plugin would look for the `.editorconfig` file in the actual project directories. Now it traverses the actual directories from the project directory up to the root project directory.

So e.g. for projects
```
: -> .
:foo -> ./projects/custom/directory/foo
:foo:bar -> ./projects/custom/directory/bar
```

then when applying the plugin in `:foo:bar` project:
 - previously we'd search for `.editorconfig` files in `./projects/custom/directory/foo`, `./projects/.../bar` and `.`
 - now, we'll search for `.editorconfig` files in `./projects/custom/directory/bar`, `./projects/custom/directory`, `./projects/custom`, `./projects/` and `.`

(fwiw I think the new behavior is more intuitive and expected)

Reply:
+ this supposed to resemble what ktlnt does when searching for _applicable_ editorconfigs, so the new behavior can be even considered a fix 👍